librz/analysis/arch/8051/8051_il.c

```diff
@@ -168,230 +270,233 @@ static RzILOpPure *borrowin(RzILOpBitVector *a, RzILOpBitVector *b, RzILOpBitVec
+		set_sub_curry(VAL_ACC, VARL("arg1"), VARL("res")),
+		i8051_reg_set(I8051_ACC, VARL("res")));
 }
 static RzILOpEffect *i_div(I8051Op *op) {
 	rz_return_val_if_fail(op && op->argv[0] && op->argv[1], NULL);
-	return SEQ6(SETL("@a", get_any(op->argv[0])), SETL("@b", get_any(op->argv[1])),
-		set_a(DIV(VARL("@a"), VARL("@b"))), set_b(MOD(VARL("@a"), VARL("@b"))),
-		set_reg(I8051_CY, IL_FALSE), set_reg(I8051_OV, IS_ZERO(VARL("@b"))));
+	return SEQ6(SETL("@a", i8051_addressing_get(op->argv[0])),
+		SETL("@b", i8051_addressing_get(op->argv[1])),
+		i8051_reg_set(I8051_ACC, DIV(VARL("@a"), VARL("@b"))),
+		i8051_reg_set(I8051_B, MOD(VARL("@a"), VARL("@b"))),
+		i8051_reg_set(I8051_CY, IL_FALSE),
+		i8051_reg_set(I8051_OV, IS_ZERO(VARL("@b"))));
 }
 static RzILOpEffect *i_mul(I8051Op *op) {
 	rz_return_val_if_fail(op && op->argv[0] && op->argv[1], NULL);
-	return SEQ6(SETL("v", DIV(UNSIGNED(16, VAL_A), UNSIGNED(16, VAL_B))),
-		SETL("mbv", HI_BYTE(VARL("v"))),
-		set_a(LO_BYTE(VARL("v"))), set_b(VARL("mbv")),
-		set_reg(I8051_CY, IL_FALSE), set_reg(I8051_OV, NON_ZERO(VARL("mbv"))));
+	return SEQ6(SETL("v", MUL(UNSIGNED(16, VAL_ACC), UNSIGNED(16, VAL_B))),
+		SETL("MSB", HI_BYTE(VARL("v"))),
+		i8051_reg_set(I8051_ACC, LO_BYTE(VARL("v"))),
+		i8051_reg_set(I8051_B, VARL("MSB")),
+		i8051_reg_set(I8051_CY, IL_FALSE),
+		i8051_reg_set(I8051_OV, NON_ZERO(VARL("MSB"))));
 }
 
 static RzILOpEffect *i_dec(I8051Op *op) {
 	rz_return_val_if_fail(op && op->argv[0], NULL);
-	return SEQ2(SETL("arg0", get_any(op->argv[0])), set_any(op->argv[0], SUB(VARL("arg0"), U8(1))));
+	return SEQ2(SETL("arg0", i8051_addressing_get(op->argv[0])), i8051_addressing_set(op->argv[0], SUB(VARL("arg0"), U8(1))));
 }
 static RzILOpEffect *i_inc(I8051Op *op) {
 	rz_return_val_if_fail(op && op->argv[0], NULL);
 	RzILOpPure *one = op->opcode == 0xa3 ? U16(1) : U8(1);
-	return SEQ2(SETL("arg0", get_any(op->argv[0])), set_any(op->argv[0], ADD(VARL("arg0"), one)));
+	return SEQ2(SETL("arg0", i8051_addressing_get(op->argv[0])), i8051_addressing_set(op->argv[0], ADD(VARL("arg0"), one)));
 }
 
 static RzILOpEffect *i_clr(I8051Op *op) {
 	rz_return_val_if_fail(op, NULL);
-	return setb_any(op->argv[0], false);
+	return i8051_addressing_set_bool(op->argv[0], false);
 }
 static RzILOpEffect *i_setb(I8051Op *op) {
 	rz_return_val_if_fail(op, NULL);
-	return setb_any(op->argv[0], true);
+	return i8051_addressing_set_bool(op->argv[0], true);
 }
 static RzILOpEffect *i_cpl(I8051Op *op) {
 	rz_return_val_if_fail(op, NULL);
-	RzILOpPure *v = (op->argv[0]->mode == I8051_ADDRESSING_REGISTER && i8051_register_is_bit[op->argv[0]->d.reg]) || op->argv[0]->mode == I8051_ADDRESSING_BIT
-		? INV(get_any(op->argv[0]))
-		: LOGNOT(get_any(op->argv[0]));
-	return set_any(op->argv[0], v);
+	RzILOpPure *v = (op->argv[0]->mode == I8051_ADDRESSING_REGISTER && i8051_reg_is_psw_bit(op->argv[0]->d.reg)) || op->argv[0]->mode == I8051_ADDRESSING_BIT
```

Comment:
Usually, for such long conditions I recommend creating a separate `static inline` function with the descriptive na,e, which will be substituted by the compiler, see [DEVELOPERS.md](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md).